web/packages/teleterm/src/ui/App.tsx

```diff
@@ -26,12 +26,16 @@ import AppContext from './appContext';
 import AppContextProvider from './appContextProvider';
 import { StyledApp } from './components/App';
 import { CatchError } from './components/CatchError';
+import { DeepLinks } from './DeepLinks';
 import { ResourcesContextProvider } from './DocumentCluster/resourcesContext';
 import { ThemeProvider } from './ThemeProvider';
 import { ConnectionsContextProvider } from './TopBar/Connections/connectionsContext';
 import { VnetContextProvider } from './Vnet/vnetContext';
 
-export const App: React.FC<{ ctx: AppContext }> = ({ ctx }) => {
+export const App: React.FC<{
+  ctx: AppContext;
+  launchDeepLink?: ComponentPropsWithoutRef<typeof DeepLinks>['launchDeepLink'];
```

Comment:
I think the use of `ComponentPropsWithoutRef` and `typeof` is warranted here. If you think about Law of Demeter in the context of React components, App should not need to know that `DeepLinks` takes as `launchDeepLink` a function where the first argument is `AppContext`, the second is `AuxContext` and so on. The only thing it needs to know that the `launchDeepLink` prop of `App` should fit as the same prop in `DeepLinks`.

Reply:
Okay, I think it's reasonable. 
Only this syntax is ugly 😆